python/samples/getting_started/evaluation/azure_ai_foundry/evaluation/self_reflection.py

```diff
@@ -0,0 +1,397 @@
+        
+        # Add to conversation history
+        messages.append(ChatMessage(role="assistant", text=agent_response))
+
+        # Request improvement
+        reflection_prompt = (
+            f"The groundedness score of your response is {score}/{max_score}. "
+            f"Explanation for score: [{feedback}]. "
+            f"Reflect on your answer and improve it to get the maximum score of {max_score} "
+            f"considering the explanation. Now please provide an updated response, taking into "
+            f"account the feedback, but make your answer sound as if it was your first response. "
+            f"Don't refer to the feedback in your answer."
+        )
+        messages.append(ChatMessage(role="user", text=reflection_prompt))
+    
+    end_time = time.time()
+    latency = end_time - start_time
+
+    # Handle edge case where no response improved the score
+    if best_response is None and raw_response is not None and len(raw_response.messages) > 0:
+        best_response = raw_response.messages[0].text
+        best_iteration = i + 1
+
+    return {
+        "best_response": best_response,
+        "best_response_score": best_score,
+        "best_iteration": best_iteration,
+        "iteration_scores": iteration_scores,  # Structured list of all scores
+        "messages": [message.to_json() for message in messages],
+        "num_retries": i + 1,
+        "total_groundedness_eval_time": total_groundedness_eval_time,
+        "total_end_to_end_time": latency,
+    }
+
+
+async def run_self_reflection_batch(
+    input_file: str,
+    output_file: str,
+    agent_model: str = DEFAULT_AGENT_MODEL,
+    judge_model: str = DEFAULT_JUDGE_MODEL,
+    max_self_reflections: int = 3,
+    env_file: str | None = None,
+    limit: int | None = None
+) -> pd.DataFrame:
+    """
+    Run self-reflection on a batch of prompts.
+
+    Args:
+        input_file: Path to input parquet file with prompts
+        output_file: Path to save output parquet file
+        agent_model: Model to use for generating responses
+        judge_model: Model to use for groundedness evaluation
+        max_self_reflections: Maximum number of self-reflection iterations
+        env_file: Optional path to .env file
+        limit: Optional limit to process only the first N prompts
+
+    Returns:
+        DataFrame with results
+    """
+    # Load environment variables
+    if env_file and os.path.exists(env_file):
+        load_dotenv(env_file, override=True)
+    else:
+        load_dotenv(override=True)
+
+    # Create agent
+    agent = AzureOpenAIChatClient(
+        credential=AzureCliCredential(),
+        deployment_name=agent_model,
+        api_key=os.environ.get("AZURE_OPENAI_API_KEY"),
+        endpoint=os.environ.get("AZURE_OPENAI_ENDPOINT"),
+    ).create_agent(
+        instructions="You are a helpful agent.",
+    )
+
+    # Load input data
+    print(f"Loading prompts from: {input_file}")
+    df = pd.read_parquet(input_file)
```

Comment:
could we make the sample resource file something readable like a csv, or json? That way you don't have to run the code to understand what's in there, if it's too large, then please add a snippet as CSV in the readme